runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogData.java

```diff
@@ -340,12 +369,17 @@ public String toString() {
      * Verify that max payload is enforced for the specified limit.
      *
      * @param limit Max write limit.
+     * @return the serialized size of the payload
      */
-    public void checkMaxWriteSize(int limit) {
-        try (ILogData.SerializationHandle sh = this.getSerializedForm()) {
-            if (limit != 0 && getSizeEstimate() > limit) {
-                throw new WriteSizeException(getSizeEstimate(), limit);
-            }
+    public int checkMaxWriteSize(int limit) {
+        Preconditions.checkState(serializedCache != null, "checkMaxWriteSize requires serialized form");
+
+        int payloadSize = getSizeEstimate();
+
+        if (payloadSize > limit) {
+            throw new WriteSizeException(payloadSize, limit);
         }
```

Comment:
This doesn't seem right. When an object is serialized (not byte array), then this always return 1. Having `serializedCache != null` doesn't imply that `LogData::data` is populated with the serialized data. 

Shouldn't this logic check the size against `serializedCache.writerIndex()` ?

Reply:
@Maithem getSizeEstimate() will use `lastKnownSize` if that is available, which will be set to the correct payload size when the payload was first serialized, so it will not return 1.